### DISCUSSION TITLE
How can i read properties from given public key?

### DISCUSSION THREAD

**Discussion by NoobjavascriptCoder** [openpgpjs/openpgpjs#1514] [Q&A] [answered]
I'm sorry for this beginner question, i'm very new to JS and OpenPGP.js.

I want to read out the properties of an given public pgp key. I already found the class "key" (https://docs.openpgpjs.org/Key.html)
And there i can find getAlgorithmInfo() for example. But how can i use the "key" class after i read the key with "const publicKey = await openpgp.readKey({ armoredKey: publicKeyArmored });"

Thank you for your support.

**Comment by larabr** [comment, marked as answer]
Hi @NoobjavascriptCoder, which information do you want to access specifically?

In your example, `publicKey` is will be a `PublicKey` instance, which is a subclass of `Key`, so you should be able to access all the methods of `Key` e.g. `publicKey.getAlgorithmInfo()`.

**Reply by NoobjavascriptCoder** [reply]
Thank you very much.
...missing the forest through the trees...
